packages/flutter/lib/src/widgets/scroll_position.dart

```diff
@@ -135,15 +133,15 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
   ///
   /// Intended to aid with identifying animation controller instances in debug
   /// output.
-  final String debugLabel;
+  final String? debugLabel;
 
   @override
-  double get minScrollExtent => _minScrollExtent;
-  double _minScrollExtent;
+  double? get minScrollExtent => _minScrollExtent;
+  double? _minScrollExtent;
 
   @override
-  double get maxScrollExtent => _maxScrollExtent;
-  double _maxScrollExtent;
+  double? get maxScrollExtent => _maxScrollExtent;
+  double? _maxScrollExtent;
```

Comment:
i think we should try making these getters non-nullable (and maybe even the fields, though they'd have to be late). it might require removing some asserts.

Reply:
Trying to make them non-nullable I faced several issue (in test) and even with an `bool get has*` method there's still the `ScrollMetrics.copyWith` issue that can not be solved.

Reply:
I found a workaround for `ScrollMetrics.copyWith`. Let me know if the `has*` pattern looks good to you.

Reply:
See ed04e18fdc

Reply:
that looks good except i would just call it "hasScrollExtents" rather than having both, since they should always be set together

Reply:
or hasDimensions since viewportDimensions could maybe also be part of it

Reply:
As there are 2 methods (`applyViewportDimension` and `applyContentDimensions`) it seems legit to have 2 get `has*`. 

IMHO if we'd like to have a single `hasDimensions` we should try to have a single  `applyDimensions` instead of the 2 `apply*` above. But regarding test failures I faced (trying to have a single `haveDimensions`) it seems to have cases where scrollExtents and viewportDimension are not all null or all non-null.

BTW should we use `hasContentDimensions` instead of `hasScrollExtents` to have the same naming as `apply*`?

Reply:
Sure, hasContentDimensions and hasViewportDimensions works for me.

Reply:
I changed `hasScrollExtents` to `hasContentDimensions`